crates/next-dev/benches/util/module_picker.rs

```diff
@@ -0,0 +1,49 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+use rand::{rngs::StdRng, seq::SliceRandom, SeedableRng};
+
+/// Picks modules at random, but with a fixed seed so runs are somewhat
+/// reproducible.
+///
+/// This must be initialized outside of `bench_with_input` so we don't repeat
+/// the same sequence in different samples.
+pub struct ModulePicker {
+    depths: Vec<usize>,
+    modules_by_depth: HashMap<usize, Vec<PathBuf>>,
+    rng: parking_lot::Mutex<StdRng>,
+}
+
+impl ModulePicker {
+    /// Creates a new module picker.
+    pub fn new(modules: &[PathBuf], prefix: &Path) -> Self {
+        let rng = StdRng::seed_from_u64(42);
+
+        let mut modules_by_depth: HashMap<_, Vec<_>> = HashMap::new();
+        for module in modules {
+            let depth = module.strip_prefix(prefix).unwrap().components().count();
```

Comment:
This is not correct. It's only the depth in the filesystem and not the depth in the import graph.